### PR TITLE
debian package setup

### LINKED ISSUE #246: .deb/.rpm files for easier installation
Hi all,

while you provide a zip file, I think a nice addition would be to provide .deb/.rpm files. If there is enough interest, I can hack a script around fpm (for a start) that can provide these formats. Additionally, if my issue with man page gets accepted, it can be packaged there as well.

Please let me know your thoughts.

### PR DESCRIPTION
**Pull request by ldemailly**
Non fpm / proper version of #302 

fixes #246 for other architectures than linux x86_64 / inclusion in standard apt repoes

### CODE REVIEW

**Comment by ldemailly** [comment]
tracking for this debian side is
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=908172

**Comment by ldemailly** [comment]
got it to work !

```
dl@ldemailly-debian-vm:~/go/src/fortio.org/fortio$ make debian-dist-test
make dist TAR=tar
make[1]: Entering directory '/home/dl/go/src/fortio.org/fortio'
echo "GIT_TAG := v1.2.2-18-g9d455f7" > release/Makefile
echo "GIT_STATUS := 7" >> release/Makefile
echo "GIT_SHA := 9d455f7c0e95be3781e3d0ebf6e6202f2f290697" >> release/Makefile
cat release/Makefile.dist >> release/Makefile
# put the source files where they can be used as gopath by go,
# except leave the debian dir where it needs to be (below the version dir)
git ls-files --recurse-submodules \
	| awk '{printf("src/fortio.org/fortio/%s\n", $0)}' \
	| (cd ../../.. ; tar \
	--xform="s|^src|fortio-1.2.2-18-g9d455f7/src|;s|^.*debian/|fortio-1.2.2-18-g9d455f7/debian/|" \
	--owner=0 --group=0 -c -f - -T -) > release/fortio_1.2.2-18-g9d455f7.orig.tar
# move the release/Makefile at the top (after the version dir)
tar --xform="s|^release/|fortio-1.2.2-18-g9d455f7/|" \
	--owner=0 --group=0 -r -f release/fortio_1.2.2-18-g9d455f7.orig.tar release/Makefile
gzip -f release/fortio_1.2.2-18-g9d455f7.orig.tar
Created /home/dl/go/src/fortio.org/fortio/release/fortio_1.2.2-18-g9d455f7.orig.tar.gz
make[1]: Leaving directory '/home/dl/go/src/fortio.org/fortio'
mkdir -p ~/tmp/fortio-dist-test
rm -rf ~/tmp/fortio-dist-test/fortio*
cp /home/dl/go/src/fortio.org/fortio/release/fortio_1.2.2-18-g9d455f7.orig.tar.gz ~/tmp/fortio-dist-test
cd ~/tmp/fortio-dist-test; tar xvfz *.tar.gz
[...]
fortio-1.2.2-18-g9d455f7/src/fortio.org/fortio/version/version.go
fortio-1.2.2-18-g9d455f7/Makefile
cd ~/tmp/fortio-dist-test;\
	ln -s *.tar.gz fortio_`cd fortio-1.2.2-18-g9d455f7; dpkg-parsechangelog -S Version | sed -e "s/-.*//"`.orig.tar.gz
cd ~/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7; dpkg-buildpackage -us -uc
dpkg-buildpackage: info: source package fortio
dpkg-buildpackage: info: source version 1.2.3-1
dpkg-buildpackage: info: source distribution UNRELEASED
dpkg-buildpackage: info: source changed by Laurent Demailly <ldemailly@gmail.com>
dpkg-buildpackage: info: host architecture amd64
 dpkg-source --before-build fortio-1.2.2-18-g9d455f7
 fakeroot debian/rules clean
make[1]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
dh clean
   dh_testdir
   dh_auto_clean
	make -j2 distclean
make[2]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
cd src/fortio.org/fortio; GOPATH=/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7 make distclean GIT_TAG=v1.2.2-18-g9d455f7 GIT_STATUS=7 GIT_SHA=9d455f7c0e95be3781e3d0ebf6e6202f2f290697
make[3]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/src/fortio.org/fortio'
rm -f /tmp/fortio_build/build-info.txt /tmp/fortio_build/link-flags.txt ../fortio.bin release/Makefile
rm -f *.profile.* */*.profile.*
rm -rf ./cert-tmp/
make[3]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/src/fortio.org/fortio'
make[2]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
   dh_clean
make[1]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
 dpkg-source -b fortio-1.2.2-18-g9d455f7
dpkg-source: info: using source format '3.0 (quilt)'
dpkg-source: info: building fortio using existing ./fortio_1.2.3.orig.tar.gz
dpkg-source: info: building fortio in fortio_1.2.3-1.debian.tar.xz
dpkg-source: info: building fortio in fortio_1.2.3-1.dsc
 debian/rules build
make[1]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
dh build
   dh_testdir
   dh_update_autotools_config
   dh_autoreconf
   dh_auto_configure
   dh_auto_build
	make -j2
make[2]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
This is the build dist for fortio v1.2.2-18-g9d455f7
please use the Makefile in src/fortio.org/fortio
make[2]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
   dh_auto_test
	make -j2 test
make[2]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
cd src/fortio.org/fortio; GOPATH=/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7 make test GIT_TAG=v1.2.2-18-g9d455f7 GIT_STATUS=7 GIT_SHA=9d455f7c0e95be3781e3d0ebf6e6202f2f290697
make[3]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/src/fortio.org/fortio'
./cert-gen
Creating test CA cert and server cert/key...
[...]
WARNING: Generated credentials are self-signed and should be used for testing purposes only.
go test -timeout 90s -race . ./ui ./echosrv ./log ./version ./stats ./bincommon ./fnet ./fcurl ./periodic ./histogram ./fhttp ./fgrpc
?   	fortio.org/fortio	[no test files]
?   	fortio.org/fortio/ui	[no test files]
?   	fortio.org/fortio/echosrv	[no test files]
ok  	fortio.org/fortio/log	(cached)
?   	fortio.org/fortio/version	[no test files]
ok  	fortio.org/fortio/stats	(cached)
?   	fortio.org/fortio/bincommon	[no test files]
ok  	fortio.org/fortio/fnet	(cached)
?   	fortio.org/fortio/fcurl	[no test files]
ok  	fortio.org/fortio/periodic	(cached)
?   	fortio.org/fortio/histogram	[no test files]
ok  	fortio.org/fortio/fhttp	33.968s
ok  	fortio.org/fortio/fgrpc	61.607s
make[3]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/src/fortio.org/fortio'
make[2]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
   create-stamp debian/debhelper-build-stamp
make[1]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
 fakeroot debian/rules binary
make[1]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
dh binary
   dh_testroot
   dh_prep
   dh_auto_install
	make -j2 install DESTDIR=/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/debian/fortio AM_UPDATE_INFO_DIR=no
make[2]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
cd src/fortio.org/fortio; GOPATH=/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7 make install GIT_TAG=v1.2.2-18-g9d455f7 GIT_STATUS=7 GIT_SHA=9d455f7c0e95be3781e3d0ebf6e6202f2f290697
make[3]: Entering directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/src/fortio.org/fortio'
rm -f /tmp/fortio_build/build-info.txt /tmp/fortio_build/link-flags.txt ../fortio.bin release/Makefile
mkdir -p /tmp/fortio_build
echo "2018-09-12 22:46 9d455f7c0e95be3781e3d0ebf6e6202f2f290697" > /tmp/fortio_build/build-info.txt
echo "-s -X fortio.org/fortio/ui.resourcesDir=/usr/share/fortio -X main.defaultDataDir=/var/lib/fortio \
  -X \"fortio.org/fortio/version.buildInfo=2018-09-12 22:46 9d455f7c0e95be3781e3d0ebf6e6202f2f290697\" \
  -X fortio.org/fortio/version.tag=v1.2.2-18-g9d455f7 \
  -X fortio.org/fortio/version.gitstatus=7" | tee /tmp/fortio_build/link-flags.txt
-s -X fortio.org/fortio/ui.resourcesDir=/usr/share/fortio -X main.defaultDataDir=/var/lib/fortio   -X "fortio.org/fortio/version.buildInfo=2018-09-12 22:46 9d455f7c0e95be3781e3d0ebf6e6202f2f290697"   -X fortio.org/fortio/version.tag=v1.2.2-18-g9d455f7   -X fortio.org/fortio/version.gitstatus=7
go version
go version go1.11 linux/amd64
CGO_ENABLED=0 GOOS= go build -a -ldflags '-s -X fortio.org/fortio/ui.resourcesDir=/usr/share/fortio -X main.defaultDataDir=/var/lib/fortio   -X "fortio.org/fortio/version.buildInfo=2018-09-12 22:46 9d455f7c0e95be3781e3d0ebf6e6202f2f290697"   -X fortio.org/fortio/version.tag=v1.2.2-18-g9d455f7   -X fortio.org/fortio/version.gitstatus=7' -o ../fortio.bin fortio.org/fortio
../fortio.bin version
1.2.3-pre 2018-09-12 22:46 9d455f7c0e95be3781e3d0ebf6e6202f2f290697-dirty go1.11
mkdir -p /home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/debian/fortio/usr/bin /home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/debian/fortio/usr/share/fortio /home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/debian/fortio/var/lib/fortio
chmod 1777 /home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/debian/fortio/var/lib/fortio
cp ../fortio.bin /home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/debian/fortio/usr/bin/fortio
cp -r ui/templates ui/static /home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/debian/fortio/usr/share/fortio
make[3]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7/src/fortio.org/fortio'
make[2]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
   dh_installdocs
   dh_installchangelogs
   dh_perl
   dh_link
   dh_strip_nondeterminism
   dh_compress
   dh_fixperms
   dh_strip
   dh_makeshlibs
   dh_shlibdeps
   dh_installdeb
   dh_gencontrol
   dh_md5sums
   dh_builddeb
dpkg-deb: building package 'fortio' in '../fortio_1.2.3-1_amd64.deb'.
make[1]: Leaving directory '/home/dl/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7'
 dpkg-genbuildinfo
 dpkg-genchanges  >../fortio_1.2.3-1_amd64.changes
dpkg-genchanges: info: including full source code in upload
 dpkg-source --after-build fortio-1.2.2-18-g9d455f7
dpkg-buildpackage: info: full upload (original source is included)
cd ~/tmp/fortio-dist-test/fortio-1.2.2-18-g9d455f7; lintian
E: fortio source: source-is-missing src/fortio.org/fortio/ui/static/js/Chart.min.js
W: fortio: binary-without-manpage usr/bin/fortio
```

File: debian/control [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Source: fortio
+Section: web
+Priority: optional
+Maintainer: Laurent Demailly <ldemailly@gmail.com>
+Build-Depends: debhelper (>= 10),
+	golang-go (>= 1.8)
```
Comment (ymesika) [inline debian/control:6]: I think `lintian (>=2.5)` should be added here.

**Review by ymesika — CHANGES_REQUESTED** [review]
Tested running this and installing the generated .deb file on my Raspbian and it works well.
Just add the `lintian` dep.

File: debian/control (hunk 1 above)
Comment (ldemailly) [reply debian/control:6]: thanks for checking! lintian is only needed for the extra test makefile target (but so are build essentials, dh*, golang etc...) not for the build itself

**Review by ymesika — APPROVED** [review]

**Pull request merged**